app/controllers/ArrivalsController.scala

```diff
@@ -21,32 +21,72 @@ import java.net.{URI, URLEncoder}
 import connectors.{ArrivalConnector, MessageConnector}
 import controllers.actions.AuthAction
 import javax.inject.Inject
-import models.request.ArrivalNotificationXSD
-import play.api.mvc.{Action, ControllerComponents}
+import models.request.{ArrivalNotificationXSD, UnloadingRemarksXSD, XSDFile}
+import play.api.mvc.{Action, ControllerComponents, Result}
 import services.XmlValidationService
-import uk.gov.hmrc.http.HttpResponse
 import uk.gov.hmrc.play.bootstrap.controller.BackendController
+import utils.Utils
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
 import scala.xml.NodeSeq
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse, Upstream4xxResponse}
 
 class ArrivalsController @Inject()(cc: ControllerComponents,
                                    authAction: AuthAction,
                                    messageConnector: MessageConnector,
                                    arrivalConnector: ArrivalConnector,
                                    xmlValidationService: XmlValidationService)(implicit ec: ExecutionContext) extends BackendController(cc) {
 
+  def createUnloadingPermission(arrivalId: String): Action[NodeSeq] = authAction.async(parse.xml) {
+    implicit request =>
+      postToArrivalEndpointDownstream(request.body.toString, UnloadingRemarksXSD, Some(arrivalId))
+  }
+
+  private def postToArrivalEndpointDownstream(body: String, xsdFile: XSDFile, arrivalId: Option[String])(implicit hc: HeaderCarrier): Future[Result] =
+    xmlValidationService.validate(body, xsdFile) match {
+      case Right(_) =>
+        messageConnector.post(body, arrivalId).map { response =>
+          response.status match {
+            case s if Utils.is2xx(s) =>
+              val location = response.header(LOCATION)
+
+              location match {
+                case Some(locationValue) => Utils.arrivalId(locationValue) match {
+                  case Success(id) =>
+                    Accepted.withHeaders(LOCATION -> s"/customs/transits/movements/arrivals/${Utils.urlEncode(id)}")
+                  case Failure(_) =>
+                    InternalServerError
+                }
+                case _ =>
+                  InternalServerError
+              }
+          }
+        } recover {
+          case e: Upstream4xxResponse =>
+            if (e.upstreamResponseCode == 400)
+              BadRequest
+            else if (e.upstreamResponseCode == 404)
+              NotFound
+            else
+              InternalServerError
+          case _: Throwable =>
+            InternalServerError
+        }
+      case Left(_) =>
+        Future.successful(BadRequest)
+    }
```

Comment:
We could change the method to take in a HTTP method, string body and xsdFile to validate against. This would mean it could get reused by all three endpoint methods.

`  sendToArrivalEndpointDownstream(request.method, request.body.toString, XSDFile, Some(arrivalId))
  }`